File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo

  COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

  def self.code(color)
    COLORS.find_index(color).to_s
  end

  def self.value(colors)
    (code(colors[0]) + code(colors[1])).to_i
  end
  
end

  
  

File: python/resistor-color-duo/resistor_color_duo.py
BANDS = {   
        "black":   0,
        "brown":   1,
        "red":     2,
        "orange":  3,
        "yellow":  4,
        "green":   5,
        "blue":    6,
        "violet":  7,
        "grey":    8,
        "white":   9 
  }
     
def value(colors):
    return 10*BANDS[colors[0]] + BANDS[colors[1]]
